src/Sylius/Behat/Page/Shop/Account/Order/IndexPage.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Page\Shop\Account\Order;
+
+use Behat\Mink\Session;
+use Sylius\Behat\Page\SymfonyPage;
+use Sylius\Behat\Service\Accessor\TableAccessorInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class IndexPage extends SymfonyPage implements IndexPageInterface
+{
+    /**
+     * @var TableAccessorInterface
+     */
+    private $tableAccessor;
+
+    /**
+     * @param Session $session
+     * @param array $parameters
+     * @param RouterInterface $router
+     * @param TableAccessorInterface $tableAccessor
+     */
+    public function __construct(
+        Session $session,
+        array $parameters,
+        RouterInterface $router,
+        TableAccessorInterface $tableAccessor
+    ) {
+        parent::__construct($session, $parameters, $router);
+
+        $this->tableAccessor = $tableAccessor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function countItems()
+    {
+        return $this->tableAccessor->countTableBodyRows($this->getElement('table'));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isOrderWithNumberInTheList($number)
+    {
+        try {
+            $rows = $this->tableAccessor->getRowsWithFields(
+                $this->getElement('table'),
+                ['number' => $number]
+            );
+
+            return 1 === count($rows);
+
+        } catch (\InvalidArgumentException $exception) {
+            return false;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRouteName()
+    {
+        return 'sylius_shop_order_index';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getDefinedElements()
+    {
+        return array_merge(parent::getDefinedElements(), [
+            'table' => '.table',
```

Comment:
We should add some meaningful ids for elements. #sylius-customer-orders or something similar would make more sense than matching any table on the page.